WcaOnRails/app/models/competition.rb

```diff
@@ -512,7 +512,7 @@ def has_event?(event)
   end
 
   def has_entry_fee?
```

Comment:
Sounds good. Since the entry fee on competition is called base_entry_fee I might call the method `has_base_entry_fee?`